### ISSUE TITLE
Move Additional Files

### ISSUE THREAD

**Issue by SonarrBot** [Sonarr/Sonarr#459]
Opt-in to move nfo, sfv, etc - we will need to keep the files named properly after a rename.

**Comment by tutipeti** [comment]
Any idea when this will be implemented? I'd like to fix my whole library to use the same naming convention everywhere, but I really don't want to rename every single subtitle file manually one by one..

**Comment by Caleb-C** [comment]
I agree, @tutipeti!  I love to explore TV shows in different languages (and accents) and find subtitles invaluable.  If I knew C# or had the time to pick it up, I would attack this one myself.  Instead of just moving the media file, it should not be a big change to move all additional files with the same name (different extension) to the same location (enabled via a setting).  This should happen when renaming existing library files or moving new files into the library.

I request, with great respect to the sacrifice of time invested by those who contribute, to address this enhancement soon.

**Comment by Taloth** [comment]
Hey folks, tnx for taking the time to reply.

Contrary to popular believe this isn't an easy feature to do 'right'.
One requirement we have is that all files are tracked in the database (this has the added advantage that it makes implementing subtitle downloading easier in the future, coz we would know which subs are already covered). So the 'move/rename base-file' trick won't satisfy that.
One might argue 'something is better than nothing', well... here's another: 'if it's worth doing, it worth doing right.' :wink:

Also, the fact that <s>xbmc</s>kodi and plex support subtitle downloading hasn't exactly made this a priority.
But that doesn't mean it's a zombie issue that never gets implemented, in fact, just a couple of weeks ago I had a chat with @markus101 about some architectural implications.

**Comment by Enf0** [comment]
The problem with having Plex (or similar tools like Subliminal) download subtitles later is that they usually guess which subtitle is the correct one for the specific release. In my experience, I often end up with out-of-sync subtitles.
The subtitles that are attached to the release is always the correct subtitles for that file.

Hoping this issue is still being looked into, it would be great to have!

**Comment by upmg** [comment]
So is this going to happen? its the only thing stopping me moving from Sickbeard to Sonarr.

**Comment by Caleb-C** [comment]
+1 !!  I wish this was implemented.  I cannot tell you how much time I spend moving the subtitles over to their destination.  Its not as simple as it seems because I have different shows on different drives (its the perfect job for a program that knows where my shows are located).

I disagree that these files would need to be tracked in order to determine which shows have subtitles.   If subtitle-downloading is implemented, it will almost certainly be done when the files are moved to their destination (or when downloading).  Therefore, if the subtitle is present at that time, then do not download it.  I do not see an advantage of keeping track of these files once at their destination (except for being able to delete them -- though this can easily be handled with an option to delete files of the same name but different extension).  Also, please keep in mind that sometimes there are several subtitle files per show (multiple languages).

**Comment by tutipeti** [comment]
+1 by me too, would love to see this finally.

**Comment by upmg** [comment]
All I need is the files not to be deleted, and renamed and moved along with the original file. If the subtitle is not present in the download I do not expect Sonarr to have to find it for me.

**Comment by Caleb-C** [comment]
Is it possible to do this with a custom script?  To do this, the script would need access to the source path (including file name) and destination path.

**Comment by bwh** [comment]
It's the year 2015 and I can't believe I'm adding my support to additional file renaming, since the functionality is not present still. :)

Seriously though, I've been using Sonarr for a long while and find it great, I tried other alternatives in the past years, but Sonarr seems the best to me. I never had the need to rename files since my downloader took care of it and my subtitle downloader mimics the file name. 

Today I renamed the files with Sonarr, I ended up with lots of files with real different names (mkv vs srt). 

If you can't add this feature soon, please at least ADD A WARNING, that only what's shown on the screen will be renamed, since many people would believe it would rename related files as well... Or an UNDO might be also lifesaving at this point.

**Comment by upmg** [comment]
Completely agree bwh, I keep checking back to see if this has been implemented yet. 

Some people want Sonarr to download the subs, I just want Sonarr not to get rid of them and rename them along with the shows filename/episode format.

We live in hope.

**Comment by dingsbums** [comment]
it's almost 2016 now and still no support for moving additional files :(

+1

**Comment by vmsteve** [comment]
+1.  I have a family member with profound hearing loss and subtitles are a necessity.   I hope someone will take this on it could help a lot of people.   It really is more than a feature it's a necessity for many.  I just need the renaming part other programs can take care of the downloading of subtitles.

**Comment by liiight** [comment]
As a developer myself, I understand the argument @Taloth made with regards to doing it right. That being said, it's a highly requested feature for a really long time.

Not to hijack this issue, but as a contributor to [Flexget](http://flexget.com/) I can offer a solution for this (using Flexget).
Using this config file with Flexget will allow using sonarr as an input for shows:
```
tasks:
  sonarr_task:
    configure_series:
      from:
        sonarr::
          base_url: '<host>'
          port: 8531
          api_key: '<api_key>'
          include_ended: false
          only_monitored: true
          include_data: yes
    rss: <url>
    move:
      to: "/filestorage2/{{series_name}}/"
      filename: '{{ series_name }} - {{ series_id }}{{ location|pathext }}'
      along:
        - sub
        - srt
```
This is just an example obviously, but using something like this would enable using Flexget alongside sonarr.

**Comment by rosskinard** [comment]
+1

After moving from SickRage to Sonarr I'm way more impressed with Sonarr and its options, but my dismay at finding all my subtitle file orphaned after the Organize was pretty big. Thousands of IDX, SUB and SRT files left and my only option is to manually rename and move them to match how Sonarr renamed the video files.

**Comment by guillaume86** [comment]
@Taloth The problem you describe seems more related to downloading subtitles that are not provided with the release. For the OP issue, maybe I'm missing something, but what about a simple option like:

    "Extensions to rename with episode": ".en.srt, .srt"

I don't think it requires storing the information somewhere else, just query that option everytime an episode is renamed and apply to files matching the previous episode name.

**Comment by EYEVO** [comment]
please add this feature subs that come with the dl are usually best

**Comment by valantislevas** [comment]
so when copy/move/rename/import extra files like .srt will be available? i saw that is fixed in develop branch?

**Comment by M0UL** [comment]
@valantislevas the devs never gives etas.

**Comment by tutipeti** [comment]
@valantislevas where did you see it on develop? As far as I see the PR #1046 is still open.

**Comment by seertenedos** [comment]
wow this has been open since 2013 and you still have not worked out how to move the subtitle files when you move the video files and to rename them. That is quite an important feature especially for shows in other languages.

**Comment by pdf** [comment]
@seertenedos if you just take look at the issue history, you can see that @markus101 has been working on this on-and-off for the last few months, so progress is definitely being made, and we can hope that this feature will land sometime in the not-too-distant future.

**Comment by seertenedos** [comment]
@pdf valid point and thanks @markus101  for working on this. I am doing this manually at the minute using FileBot. Sonarr is such a great app and i have been using it for years but i never realised this was missing as most of the subs i got were embedded. Really looking forward to this being completed

**Comment by haudankaivajasi** [comment]
Hey. What is the situation of this?

**Comment by markus101** [comment]
Parsing of files (and storing them in the DB) has been merged in. Importing and renaming files will come in future releases, we want to ensure the parsing is working properly first.

**Comment by casperse** [comment]
So Sonarr will still delete all my subtitles (Would be happy just to get them moved to the renemed folder)
Can anyone confirm if its true that sickrage can do this? (Need country subtitles)

**Comment by valantislevas** [comment]
yes i use sickrage only for post proccessing the series (rename/move).. untill sonarr is able to do

**Comment by casperse** [comment]
I have played around with both, and why not just use sickrage for everything? (Here I can also create a filter pr tv show, and not as in Sonarr one filter "words" for all? Please help me understand what sonarr can do better (I am just now testing them all, and would like some input) Thanks

**Comment by markus101** [comment]
Please keep this thread on topic, if you wish to discuss the differences there are better mediums.

**Comment by Waseh** [comment]
So i just tested this and im getting filenames that are missing a dot (".")
Instead of file.1080p.en.srt i get file.1080p.ensrt, and instead of  file.1080p.nfo-orig i get file.1080pnfo-orig.
Have i setup something wrong or is there an issue with the renaming?

**Comment by markus101** [comment]
Thanks for reporting @Waseh I found the cause and pushed a fix for it, it'll be in the next build.

**Comment by spooky69** [comment]
Great feature, hope to get it soon.

**Comment by valantislevas** [comment]
when we can use that feature on master repo? @markus101

**Comment by markus101** [comment]
We can't provide an ETA, we've been fixing a few things with it, but it shouldn't be too much longer.

**Comment by tenzion** [comment]
Really looking forward to this feature!

**Comment by sim099** [comment]
This feature doesn't seem to work for me, at least not for nfo files. I enabled it straight after the latest update was applied and forgotten about it. Going back and checking now, my custom script that simply copies extra files into another directory has nfo files for the last few that I've imported. I've checked the last few downloads in the target directory (i.e. my NAS drive) and none of them have "nfo-orig" files. I haven't imported anything with subtitles since then though, so can't say if it's just nfo files or not.

I have "Import Extra Files" set to "Yes" and "Extra File Extensions" to "srt,sub,idx,nfo". Is this correct? Wasn't entirely sure whether to include the full stops, but ended up going by the example from the tooltip of the help icon, which is "sub,nfo".

Is there any other settings that I may not have configured correctly?
